### PR TITLE
Add support for rescheduled pods with the same name in drain

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Add support for rescheduled pods with the same name in drain.

### CODE REVIEW

File: cluster-autoscaler/core/scale_down.go [hunk 1]
```diff
@@ -763,12 +763,12 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
 		allGone = true
 		for _, pod := range pods {
 			podreturned, err := client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
-			if err == nil {
+			if err == nil && podreturned.Spec.NodeName == node.Name {
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_down.go]: I will feel better if you add `podreturned != nil` check. It should probably never happen without some err, but it's still a pointer.

File: cluster-autoscaler/core/scale_down.go [hunk 2]
```diff
@@ -763,12 +763,12 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
...
 				glog.Errorf("Not deleted yet %v", podreturned)
 				allGone = false
 				break
 			}
-			if !kube_errors.IsNotFound(err) {
+			if err != nil && !kube_errors.IsNotFound(err) {
 				glog.Errorf("Failed to check pod %s/%s: %v", pod.Namespace, pod.Name, err)
 				allGone = false
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_down.go]: Since you're already working on this file - shouldn't there be a break here?

Comment (aleksandra-malinowska) [reply cluster-autoscaler/core/scale_down.go]: Done

**Comment by MaciekPytel** [comment]
/lgtm
Please merge once you're done testing. Also I think this is worth cherry-picking.

**Comment by mwielgus** [comment]
Is the testing finished?

**Comment by aleksandra-malinowska** [comment]
Sorry, found (and fixed) another issues with taints persisting through autoscaler restarts. 

Tested manually that StatefulSet pods are now correctly recognized as being moved to another node.

**Pull request merged**